gridstatus/caiso.py

```diff
@@ -268,10 +298,49 @@ def _process_solar_and_wind_forecast_dam(self, data):
                     "Wind": "Wind MW",
                 },
             ),
-            ["Interval Start", "Interval End", "Location"],
+            ["Interval Start", "Interval End", "Publish Time", "Location"],
         )
 
-        return df.sort_values(["Interval Start", "Location"]).reset_index(drop=True)
+        return df.sort_values(
+            ["Interval Start", "Publish Time", "Location"],
+        ).reset_index(drop=True)
+
+    def _add_forecast_publish_time(self, data, publish_time_offset_from_day_start):
+        """
+        Labels forecasts with a publish time using the logic:
+
+        - If tomorrow or further in the future, the publish time is
+            * Today's publish time if current time is after the publish time
+            * Yesterday's publish time if current time is before the publish time
+        - If today or earlier, the publish time is the previous day's publish time
+
+        We assume the forecast was published the day before the forecasted day unless
+        the forecast is in the future to avoid having publish times in the future.
+        """
+        current_time = pd.Timestamp.now(tz=self.default_timezone)
+        todays_publish_time = (
+            current_time.normalize() + publish_time_offset_from_day_start
```

Comment:
i dont think adding Timedeltas works with day light savings time 

<img width="852" alt="image" src="https://github.com/kmax12/gridstatus/assets/371792/8b69fbe9-b7db-463e-bcfa-14de6c98d960">

in this case, i think a replace would work. there may be some other ways
<img width="800" alt="image" src="https://github.com/kmax12/gridstatus/assets/371792/4f9172fc-243e-4d24-be6d-9fb8d25361a9">